### ISSUE TITLE
prefer-const rule not work at `.vue` file

### ISSUE THREAD

**Issue by daiwanxing** [vuejs/eslint-plugin-vue#2132]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: latest**  
- **eslint-plugin-vue version: latest** 
- **Node version: 16.10.0** 
- **Operating System: windows 11** 

**Please show your full configuration:**

```json5

```

**What did you do?**	

```vue
// detect this code and report warning or error.
var foo = 'foo'; 
```

I would like give a [live demo](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJyNV82OI7cRfhVGPowGGGqALBA42tlJAtuHHAIEiJGLJwd2d7VEL5vsNNkaKYN5Cl/9dH4Sf0X2f0ubvcyoq4rF+vnqh2+b3BW02W+eAlW1UYGeX6wQT9rWbRAnWYFrPr1s4ve/lWnpZSMek0zWhuCs+GtudP4ZMkdlC0Pf8ReEcqO8B1UVhUySL5vnvxWF+NF9754eEykpas0gHVzhpNE+QJh54BqdfgiYU7oGQluWehDaFnS+xz/B33+HA/5l08vuP9Ol07fTxUifXaRxZnKk572Ju4EtZeEs3e3jHTv+Ld7HI4PvI/eT+MPw0Qs+9/JPvlYWfuhyega+/vbrL0+PzHwWb2/prkDnIN7fuyg8Gp2C9dga/Hh6nOTryeeNroPwFNoa37qqXRPEm2iofBC5q5A7KsS7KBtXiZfNqY2W5c76IMbMwnQc2IK1ue+5Q2Q75k//AevFlq3Ng0byJznf3os3tnA4sjux1l3d+uP2Lbmhi734hwrHXYNzrtre74L7V2i0PWw//Ol+56GHtvLb+4ckzsHZi1IZTx2Fg7Kf2JzuiMz3e/67ZMFudujFvrPdJ9WI0jkQ7348UvwJklaZIaG9KAgQaBCq1sMkESDCJwClV9cUAjYLf3StKRCLGGJqGteI7CKMtoGaBwFYi0IXwrpwhzA595GvTbFMkP/OGRz5NKRli7h9ehbbleF/geGZUbGY9vj9egQekZr7j3OF/3Tx7h9OZINfKb4aE1atrQ5amU657dDKygHEiKeIrHAxDLHdWMYpyzm7sQeUM5ThduIaaxCiTkZJilYNctNGAivuOiPuYMMdm3AH/z7GXO2GTpDu418yWrMXLDmV4jpNUhnCdWhcawvZGfgNUbmUTTV988SHMvvztx8+Trz8piw7JYhNF5HNw6ZpDfnN/o0L6hFhr+CrLHRDqI0TmuofHyLnZ3+WrScvgSU/UKvWBC0ZV5IzBoNw2KqKRhHrpALAXmVfbl5qK19VyI8zGX+xOXP6zMu6cTU1Qc915Udt+DIkxoYZY32O1RUqqKlYQTVcUyzILOmyn+GpTEWj2L5b0uRD4y74ZXRJ+SU3dEvSGeiKaSdLjZeq1l8W9TlMLrxx4UvCCYZfECi1AVxvMbvwKyP7rntD8hgqI8lQRIL2N6S0Rbug/6eLk+FBKBWQIsPxpjqgl0sNOj/LoA7x4A3RGC2pAlouSvbWzRzNrxP6On2p+CXD9KbETe9O0lnUBVcUbyO61DfzlETbKkM0etGbalviUij1QaK98xI0l2zhFxhr4gn59YQJvuBgdEHxGIvZSTrzuOCiSq1VxlG9lIBaVXIq1avSswod6uaWQNUGhMim5M+urtF0mBFH1ZSBcY7WIbniMdjmHE/Nib7YmAaZZZAGxsqS6PMAa7SENg9tMwONP/L4ndy77EFeFySpRDSC/4qW15cYWzmnnwNuUoibDyo/zoxoLbr1xAh/hTlr5Uz2tLLlgrR9BmgpHF2xFOYdUL7qcJRxq12wMVf8aPx1TMVzMw8Z/rHJz8XYWfyLvcnZ0a+pVBwqV8GFiilBTDul5A2Sa2dgN/TfFhNvkrJJJfdMDogMlzqWXEr8xOVeqiEs9A3+ARV2xe1avB+H4UokuMMBJaJtREkArL2ejaVeMIVuiomBo4wuhqY766TJrmugu1yRoUoHn9SpqbMp90gDnVU+RjndOyQ61tuCCaN4akTF11nzikssy8kPWM8X9AE3jXNLM1K/XhFz49RSzbobTukr4hTpPY1xuSJeUaiXruF1SpVbE/nJuqJOUDCS8rWFdbOmYed/XRPNlbhxqUXiBOfSY2wktxl13X7Y2XOgEOaYHxY6ZLPSiC2dKW97CPG2pkCcLF1+YPDzFYe9PgyqGKpodIDBQf6PGjfSMZCXojD5Gk2hqFl+rKXEwU5nYPmBzj2xoKw9HKgZvw0FLqqRwiNUNQc/IyTLkc2sH9g9Z4qu1VieDV4OS08Ftrui7D/QFxGunGMdL5sza8VZGHyj8yIMyGqjZOacIWX5pjDneBRlT8Fr1YQjdrLDcSChYy00HozLsE7OiV2PXRAtFuGr+dY2wS6moB6oDb5bXo3jg9HXKh/iYhzC7EoGea79JJsAGlwreE+4EahKn3npY23AGMATVDYwLb1iu6syZwaKsy4P8BCG6wr/yeeqHgzB0wFhNGbQEIX7DyA/uDgushZ7uR5dxnKQnugjAb53ZvU0jM9yEUR/rcywbBR4XDW8iuic+/m45TCbNycGa6NGvPEaLjNCIdOkruNegCY8fiAhFFXGZx6v+yMPeweCnJkJyauS8AbBC8NcFlRUrpoWXkd2dfcigSq8ou1h5McFxaiMxvAut5bJlsEP4DjiyY5A6Zn58NacEOep5B0m/u5H6EVTt4HwrNPeqq71prmDtLoUqG6ziD0GhPf33wGEWB2e) link which from [eslint-plugin-vue](https://eslint.vuejs.org/) provided demo site.

As you can see from the demo, I used `var` keywords and declared a variable which named `foo`, and then I enabled `prefer-const` rule,  This code should has been detected error by linter, But actually It didn't.

![image](https://user-images.githubusercontent.com/37327614/232265394-69a893ba-edc0-45f5-856b-c07ce48e0662.png)

please give me a solution, Thanks!

**What did you expect to happen?**

Report error or warning by rule params

**What actually happened?**

not change

**Repository to reproduce this issue**

**Comment by ota-meshi** [comment]
The rule does not report var.  You should also use no-var.
https://eslint.org/docs/latest/rules/prefer-const

**Comment by daiwanxing** [comment]
Thanks for your suggestion, Now It's work fine.
